### PR TITLE
🔧 Assert TMC slave addresses, minor pins cleanup

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

Assert TMC slave addresses for boards with integrated TMC drivers at compile-time & some minor pins cleanup.

### Requirements

Any of the affected motherboards in this PR.

### Benefits

TMC slave addresses will be checked for validity at compile-time.

### Related Issues

None. Found while while helping someone on Discord who was following an incorrect tutorial.

### CODE REVIEW

File: Marlin/src/pins/gd32f1/pins_SOVOL_V131.h [hunk 1]
```diff
@@ -62,6 +62,28 @@
...
+    static_assert(Z_SLAVE_ADDRESS == 3, "Z_SLAVE_ADDRESS must be 3 for BOARD_SOVOL_V131.");
+  #else
+    #define Z_SLAVE_ADDRESS                    3
+  #endif
+  #ifdef E0_SLAVE_ADDRESS
+    static_assert(E0_SLAVE_ADDRESS == 3, "E0_SLAVE_ADDRESS must be 3 for BOARD_SOVOL_V131.");
+  #else
+    #define E0_SLAVE_ADDRESS                   3
+  #endif
+
```
Comment (thisiskeithb) [inline Marlin/src/pins/gd32f1/pins_SOVOL_V131.h:82]: Pulled from https://github.com/hillsoftware/sv06/blob/00d5ee302758703c9a959309716b5f720d750afc/Marlin/Configuration_adv.h#L3030-L3044

**Comment by thisiskeithb** [comment]
I've reworked this PR to follow the same format as what was done in https://github.com/MarlinFirmware/Marlin/commit/02691060664b1d582ed72c207d0062bd7e5813d4.

File: Marlin/src/pins/stm32g0/pins_BTT_EBB42_V1_1.h [hunk 2]
```diff
@@ -105,8 +106,9 @@
 
   // Default TMC slave addresses
   #ifndef E0_SLAVE_ADDRESS
-    #define E0_SLAVE_ADDRESS 0b00
+    #define E0_SLAVE_ADDRESS               0b00
   #endif
+  static_assert(E0_SLAVE_ADDRESS == 0b00, "E0_SLAVE_ADDRESS must be 0b00 for BOARD_BTT_EBB42_V1_1.");
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32g0/pins_BTT_EBB42_V1_1.h:111]: It is rather odd that some are defined in decimal and others in binary.

Comment (thisiskeithb) [reply Marlin/src/pins/stm32g0/pins_BTT_EBB42_V1_1.h:111]: I don't know why they are that way, but I kept that format to not break anything.

File: Marlin/src/pins/gd32f1/pins_SOVOL_V131.h [hunk 3]
```diff
@@ -62,6 +62,24 @@
...
+  #ifndef Z_SLAVE_ADDRESS
+    #define Z_SLAVE_ADDRESS                    3
+  #endif
+  #ifndef E0_SLAVE_ADDRESS
+    #define E0_SLAVE_ADDRESS                   3
+  #endif
+  static_assert(X_SLAVE_ADDRESS == 3, "X_SLAVE_ADDRESS must be 3 for BOARD_SOVOL_V131.");
+  static_assert(Y_SLAVE_ADDRESS == 3, "Y_SLAVE_ADDRESS must be 3 for BOARD_SOVOL_V131.");
+  static_assert(Z_SLAVE_ADDRESS == 3, "Z_SLAVE_ADDRESS must be 3 for BOARD_SOVOL_V131.");
+  static_assert(E0_SLAVE_ADDRESS == 3, "E0_SLAVE_ADDRESS must be 3 for BOARD_SOVOL_V131.");
```
Comment (sjasonsmith) [inline Marlin/src/pins/gd32f1/pins_SOVOL_V131.h:81]: Address 3? That's weird. I'm just making a note to double-check this.

File: Marlin/src/pins/stm32f1/pins_FYSETC_AIO_II.h [hunk 4]
```diff
@@ -24,13 +24,13 @@
...
 #define BOARD_INFO_NAME   "FYSETC AIO II"
-#define BOARD_WEBSITE_URL "fysetc.com"
+#define BOARD_WEBSITE_URL "github.com/FYSETC/FYSETC-AIO_II"
 
 #define BOARD_NO_NATIVE_USB
 #define RESET_STEPPERS_ON_MEDIA_INSERT
 #define DISABLE_JTAG
 
-#define pins_v2_20190128                          // new pins define
+#define PINS_V2_20190128                          // new pins define
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f1/pins_FYSETC_AIO_II.h:33]: I was wary of you changing this in case people are overriding it...but it is not set up to be overridden.

I'm guessing this is just clutter from some early prototypes or something... If that is the case ideally we could just deleted it and all the code that it was hiding.

File: Marlin/src/pins/stm32f4/pins_FYSETC_CHEETAH_V30.h [hunk 5]
```diff
@@ -41,12 +41,12 @@
...
-//  #define SERVO0_PIN                      PA0
-//#elif !defined(Z_MIN_PROBE_PIN)
-//  #define Z_MIN_PROBE_PIN                 PA0
-//#endif
+#if ENABLED(BLTOUCH)
+  #error "You need to set jumper to 5V for BLTouch, then comment out this line to proceed."
+  #define SERVO0_PIN                        PA0   // PROBE
+#elif !defined(Z_MIN_PROBE_PIN)
+  #define Z_MIN_PROBE_PIN                   PA0
+#endif
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f4/pins_FYSETC_CHEETAH_V30.h:49]: Do you know why this was commented out before? I assume something it it was incorrect or problematic?

It seems unrelated to the rest of the changes.

File: Marlin/src/pins/gd32f1/pins_SOVOL_V131.h (hunk 3 above)
Comment (sjasonsmith) [reply Marlin/src/pins/gd32f1/pins_SOVOL_V131.h:81]: I verified it in their config in the SV06 repo:

https://github.com/Sovol3d/Sv06-Source-Code/blob/main/Marlin/Configuration_adv.h

Comment (thisiskeithb) [reply Marlin/src/pins/gd32f1/pins_SOVOL_V131.h:81]: See my comment above as well: https://github.com/MarlinFirmware/Marlin/pull/26842#discussion_r1554713169

File: Marlin/src/pins/stm32f1/pins_FYSETC_AIO_II.h (hunk 4 above)
Comment (thisiskeithb) [reply Marlin/src/pins/stm32f1/pins_FYSETC_AIO_II.h:33]: It's been there since day one of the pins file, so I'm not sure why it exists.

File: Marlin/src/pins/stm32f4/pins_FYSETC_CHEETAH_V30.h (hunk 5 above)
Comment (thisiskeithb) [reply Marlin/src/pins/stm32f4/pins_FYSETC_CHEETAH_V30.h:49]: This isn't the first time we've seen quirkiness in FYSETC pins and instead of removing the disabled defines, I pulled up their board info and verified / enabled these defines and added PCB silkscreen labels as comments like we usually do).

**Review by sjasonsmith — APPROVED** [review]
This all seems in line with the style thinkyhead used previously in commit 02691060664b1d582ed72c207d0062bd7e5813d4.
